src/App.tsx

```diff
@@ -1,8 +1,44 @@
-import React from "react";
-import "./css/App.css";
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import './css/App.css';
+import {
+	addAction,
+	minusAction,
+	handleInputChange,
+	multiplyAction,
+} from './redux/actions';
+import { AppState } from './redux/types';
 
 function App() {
-  return <div></div>;
+	const { count, input } = useSelector((state: AppState) => state.counter);
+	console.log('count :', count);
+	console.log('input', input);
+	const dispatch = useDispatch();
+
+	const handleAdd = () => dispatch(addAction());
+	const handleMinus = () => dispatch(minusAction());
+	const handleMultiply = () => dispatch(multiplyAction(input));
+
+	return (
+		<>
+			<section>
+				<p>Result: {count}</p>
+			</section>
+			<section>
+				<button onClick={handleAdd}>ADD 1</button>
+				<button onClick={handleMinus}>MINUS 1</button>
+			</section>
+			<section>
+				<input
+					type='number'
+					placeholder='Multiply result with . . .'
+					onChange={(e: any) => dispatch(handleInputChange(e.target.value))} // TODO: why ((event: React.ChangeEvent<HTMLInputElement>) => was not helpful here???
```

Comment:
I believe you can safely remove **any** for `e` as typescript can infer the type itself

Reply:
I just notice, maybe you use **any** because you have problem with typescript saying the parameter does not match. Because in your action, the parameter of `handleInputChange` has the type of number. However, remember that `e.target.value` always has the type of string, even though you specify that input type is text. So you need to convert that target value before passing to the action

Reply:
Awesome suggestion. Thanks